mimesis/enums.py

```diff
@@ -0,0 +1,23 @@
+from enum import Enum
+from random import choice
+
+
+class PortRange(Enum):
+    DEFAULT = (1, 65535)
+    WELL_KNOWN = (1, 1023)
+    EPHEMERAL = (49152, 65535)
+    REGISTERED = (1024, 49151)
+
+
+class Gender(Enum):
+    FEMALE = 'female'
+    MALE = 'male'
+    F = FEMALE
+    M = MALE
+    RANDOM = choice([FEMALE, MALE])
+
+
+class TitleType(Enum):
+    TYPICAL = 'typical'
+    ACADEMIC = 'academic'
+    RANDOM = choice([TYPICAL, ACADEMIC])
```

Comment:
The same here.

Reply:
@sobolevn Do you have any suggestions, how to get random value each time?

Reply:
Maybe we can try to use `lambda` for that?

Reply:
@sobolevn I have tried this, but it's broke type hints and looks not so good, although it works:

```python
class TitleType(Enum):
    # ...
    RANDOM = lambda: choice(['typical', 'academic'])
```

```python
for i in range(5):
    print(TitleType.RANDOM())

'typical'
'academic'
'typical'
'academic'
'academic'
```

Reply:
We can just return string `random` and choice random value in method:

```python
  if title_type and title_type in TitleType:
            if title_type == TitleType.RANDOM:
                title_type = self.random.choice(
                    [TitleType.ACADEMIC, TitleType.TYPICAL]
                )
            titles = titles[title_type.value]
```
Here i'm not sure too.

Reply:
@sobolevn Do you have any better solution in mind?

Reply:
So, i have created question on [SOF](https://stackoverflow.com/questions/47353555/how-to-get-random-value-of-attribute-of-enum-on-each-iteration).

Reply:
@sobolevn I like this [solution](https://stackoverflow.com/a/47354435/5461570). What do you think about one?

Reply:
I like this one:

```python
import random
import enum
class RANDOM_ATTR(enum.EnumMeta):
    @property
    def RANDOM(self):
        return random.choice(list(self))


class Gender(enum.Enum,metaclass=RANDOM_ATTR): #this syntax works for python3 only
    FEMALE = 'female'
    MALE = 'male'


print(Gender.RANDOM)
```

Results:

```
>>> print(Gender.RANDOM)
Gender.MALE
>>> print(Gender.RANDOM)
Gender.MALE
>>> print(Gender.RANDOM)
Gender.FEMALE
>>> print(Gender.RANDOM)
Gender.FEMALE
>>> print(Gender.RANDOM)
Gender.FEMALE
```

Reply:
It could even be a simple mixin, not a metaclass.

Reply:
The problem with this solution  is that we have to many `enums` where we need random choice. We can't create metaclasses for each ones, it will looks dirty.

Reply:
We can inherit this class:
```python
class Enum(enum.Enum, metaclass=enum.EnumMeta):

    RANDOM = None
    @classmethod
    def get_random(self):
        return choice(list(self))
```
Use it like that:
```python
def check_gender(gender: Gender = Gender.RANDOM) -> Gender:
    if not gender:
        return gender.get_random()

    elif gender and gender in Gender:
        return gender
    else:
        raise UnexpectedGender(
            'You should use "Gender" from the module mimesis.enums',
        )
```

Reply:
I think that this is a good compromise.

Reply:
Seems like my solution does not work correctly, sorry.

Reply:
I take my words back.

Reply:
@sobolevn It's can create some problems with type-hints. Should we use annotation like this `Union[Gender, property]`? Because `RANDOM` is property.

Reply:
@sobolevn That's why i ask this question:
![image](https://user-images.githubusercontent.com/15812620/32962497-c1fcda74-cbdd-11e7-8a00-2491b22161c2.png)

Reply:
Oh, I was not aware of that. That's a problem. https://github.com/python/mypy/issues/220

Reply:
@sobolevn I don't quite understand how it works, but this descriptor fix this issue:

```python
class ClassPropertyDescriptor(object):
    """Author of this class is Mahmoud Abdelkader
    """
    def __init__(self, fget, fset=None):
        self.fget = fget
        self.fset = fset

    def __get__(self, obj, klass=None):
        if klass is None:
            klass = type(obj)
        return self.fget.__get__(obj, klass)()

    def __set__(self, obj, value):
        if not self.fset:
            raise AttributeError("Can't set attribute")
        type_ = type(obj)
        return self.fset.__get__(obj, type_)(value)

    def setter(self, func):
        if not isinstance(func, (classmethod, staticmethod)):
            func = classmethod(func)
        self.fset = func
        return self


def classproperty(func):
    if not isinstance(func, (classmethod, staticmethod)):
        func = classmethod(func)
    return ClassPropertyDescriptor(func)
```